### PR TITLE
KV postgres use advisory lock while performing table creation

### LINKED ISSUE #5196: Docker compose using postgres and number of replicas fail with error on relation creation
While using docker-compose with lakefs configured to use 3 replicas and postgres database, startup fail with the following error:

```
setup failed: ERROR: relation \"kv_49\" already exists (SQLSTATE 42P07)
```

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #5196

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]
LGTM, thanks for fixing quickly!

File: pkg/kv/postgres/store.go [hunk 1]
```diff
@@ -152,26 +153,45 @@ func parseStoreConfig(runtimeParams map[string]string, pgParams *kvparams.Postgr
...
-func setupKeyValueDatabase(ctx context.Context, conn *pgxpool.Conn, table string, partitionsAmount int) error {
+func setupKeyValueDatabase(ctx context.Context, conn *pgxpool.Conn, table string, partitionsAmount int) (err error) {
+	var aid string
+	aid, err = generateAdvisoryLockID("lakefs:" + table)
+	if err != nil {
+		return err
+	}
+
+	// This will wait indefinitely until the lock can be acquired.
+	_, err = conn.Exec(ctx, `SELECT pg_advisory_lock($1)`, aid)
```
Comment (N-o-Z) [inline pkg/kv/postgres/store.go:164]: @nopcoder can we add a lock_timeout and provide a detailed error message.
We should want to know when a lock was acquired and not released - and add details of required action on the event of.

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Thanks! One comment

File: pkg/kv/postgres/store.go (hunk 1 above)
Comment (nopcoder) [reply pkg/kv/postgres/store.go:164]: @N-o-Z This is the same technique we used when `go-migrate` was used. (https://github.com/golang-migrate/migrate/blob/dc26c41ac9dac4a1023c87723ffda644e41af1fe/database/pgx/pgx.go#L226)
The lock timeout is specified in the the database configuration (lock_timeout) the default is unlimited.

**Review by N-o-Z — APPROVED** [review]
Thanks for addressing the comment!

**Pull request merged**